src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnector.java

```diff
@@ -129,39 +130,39 @@ public void test(FailureCollector failureCollector) throws ValidationException {
   public BrowseDetail browse(BrowseRequest browseRequest) throws IOException {
     BigQueryPath path = new BigQueryPath(browseRequest.getPath());
 
-    BigQuery bigQuery = getBigQuery();
-    if (path.isRoot()) {
+    String dataset = path.getDataset();
+    if (dataset == null) {
       // browse project to list all datasets
-      return listDatasets(bigQuery, browseRequest.getLimit());
+      return listDatasets(getBigQuery(config.getDatasetProject()), browseRequest.getLimit());
```

Comment:
A little confused here, there are couple places `getBigQuery` is called, sometimes it uses project, sometimes it uses dataset project, what is the difference and how do we decide which project to use?

Reply:
as commented in getBigQuery method, it's where BQ list dataset and where the BQ job is run. 

So to list dataset, we should use `dataset project`. List tables already have dataset set in the tableId, so for other call we just use `project`

Reply:
also add a javadoc for that method to make it more clear